Dockerfile

```diff
@@ -33,7 +33,7 @@ RUN apt update \
          python-is-python3 \
          python3-numpy python3-matplotlib python3-scipy python3-shapely python3-packaging \
          # Additional installs
-         python3-pip wget git vim \
+         python3-pip wget git vim libspatialindex-dev \
```

Comment:
@cirrusasf are you sure we need this? the `rtree` python package page implies that it bundles `libspatialindex` in the wheel and we should only need to pip install `rtree`:
![image](https://github.com/ASFHyP3/hyp3-gamma/assets/7882693/32035a89-0ffb-41b4-9087-7c91292b863c)
https://pypi.org/project/Rtree/

And in the docker build, I see a wheel being installed:
https://github.com/ASFHyP3/hyp3-gamma/actions/runs/6727289790/job/18284909084#step:8:3611

Reply:
Joe, since you have already approved it. Does it mean we still keep the libspatialindex-dev in the Dockerfile and use conda to install rtree? I have merged the PR. If we need otherwise, please let me know.